web_src/js/features/repo-code.js

```diff
@@ -120,6 +121,47 @@ function showLineButton() {
   });
 }
 
+function postProcessPackageJson() {
+  let data;
+  try {
+    data = JSON.parse(getFileViewContent());
+  } catch {
+    return;
+  }
+
+  const packages = new Set();
+  for (const key of [
+    'dependencies',
+    'devDependencies',
+    'optionalDependencies',
+    'peerDependencies',
+  ]) {
+    for (const packageName of Object.keys(data?.[key] || {})) {
+      packages.add(packageName);
+    }
+  }
+
+  // match chroma NameTag token to detect JSON object keys
+  for (const el of document.querySelectorAll('.code-inner .nt')) {
```

Comment:
Why so hacky when you already parsed the JSON file?
Instead, you can simply check if `data['dependencies', 'devDependencies', …]` exists.
Should work without these Chroma hacks…

Reply:
The problem is we need to find the right place to do the change in the DOM, and if you look at it, it's just a soup of syntax tags, nothing we can parse by only looking at HTML.

Reply:
Ah, you mean to understand where to insert the link?

Reply:
Yes, `packages` holds the object keys to alter, but there seems to be no easy way to find the right place. Maybe by parsing the JSON as text and saving the byte offset it could be made more precise, but also more prone to changes in the HTML structure.